components/state.js

```diff
@@ -0,0 +1,26 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var states = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
+ 
+   if ( arguments.length === 1 || arguments.length === 3 ) { 
+       if ( _.isUndefined( states[inportName] ) ) { 
+           return;
+       } 
+       return states[inportName];
+   }
+
+   if ( arguments.length === 2 || arguments.length === 4 ) { 
+       states[inportName] = { data: arguments[arguments.length-1],
+                              lm: Lm(),
+                              previousLms: [],
+                              port: inportName };
```

Comment:
The port object is the port that owns that state.  Maybe it isn't needed.  I was thinking that it would allow the Framework to easily determine, for example, where messages should be sent if the state changes.  But since we do not have a known need for it yet, let's leave it out for now.  And maybe we'll find that we never need it.